frontend/packages/dev-console/src/components/import/git/GitSection.tsx

```diff
@@ -64,13 +64,10 @@ const GitSection: React.FC<GitSectionProps> = ({ showSample }) => {
       setFieldValue('git.isUrlValidated', false);
       setFieldError('git.url', 'Git repository is not reachable.');
     }
-
-    validateForm();
```

Comment:
This is fix a temporary solution or a permanent one? Removing functionality like this for our interim hack may be undesirable. I suppose we could just revert this commit if Formik provides an actual solution to this issue.

Reply:
I've removed use of `validateForm` from here because it was not doing anything productive anyway. Even after calling `validateForm`, the validation was running on old stale data which was causing the validation errors. Plus, the custom hook that i've added calls this same method but on the change of data and not right after `setFieldValue` which is an async action.

Reply:
I didn't have an issue with the other spots this change happened, because you were using the power of hooks to apply additional functionality additively. This however is you replacing current functionality with this new hook.

Is this hook something we intend to keep using? Or are we only doing this because of typing issues in upgrading. I assume Formik is fixing their own issues and this is not an issue on the latest Formik version, and we are just unable to upgrade due to the dependency cascade issue in the React types.

My only concern here is that we will have to take care to bring back the functionality as it were once we upgrade. We lack the proper safety net (I believe) that comes with unit tests & integration tests to be certain we will be safe with changes.

Reply:
Latest version of Formik also doesn't fix the issue, Formik team is still in the process of figuring out the solution. So, its not an issue that can be fixed by just upgrading to the latest Formik. And use of `validateForm` here was redundant and not needed here since every time we call `setFieldValue` or `setFieldTouched`, formik also runs the validation. So, once the problem in Formik gets resolved this will automatically start to work without the need of extra call to `validateForm` or the custom hook. Unless they hange something in the API which will need us to update the code in some way anyways.

Reply:
Ah I see - I read your comments from top-down on your PR. I apologize for my ignorance :)